traveler-ios-ui/TravelerKitUI/ViewControllers/CatalogItemViewController.swift

```diff
@@ -54,7 +52,14 @@ open class CatalogItemViewController: UIViewController {
 
         performSegue(withIdentifier: "loadingSegue", sender: nil)
 
-        Traveler.fetchCatalogItemDetails(catalogItem, delegate: self)
+        switch catalogItem {
+        case let catalogItem as ProductItem:
+            Traveler.fetchCatalogItemDetails(catalogItem, delegate: self)
+        default:
+            Log("Unknown CatalogItem Type", data: nil, level: .error)
+            performSegue(withIdentifier: "errorSegue", sender: nil)
```

Comment:
You have already performed loading segue...this doesnt break but we should avoid making multiple perform calls in the same run loop

Reply:
Where else am I performing loading segue?. If I'm not mistaken, this is the only place in this VC that loading segue is called.

Reply:
unresolved